Return a message instead of void from rpc udf

satgera opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25283 Return a message instead of void from rpc udf**

Return a message instead of void from rpc udf
This is to help thrift style rpc where there is no need for explicit send for a response.
We need to figure out how to solve the non-blocking callback case but don't want to block the thrift backed rpc agent implementation till then.

Differential Revision: [D16825072](https://our.internmc.facebook.com/intern/diff/D16825072/)

satgera committed:
Return a message instead of void from rpc udf

Return a message instead of void from rpc udf
This is to help thrift style rpc where there is no need for explicit send for a response.
We need to figure out how to solve the non-blocking callback case but don't want to block the thrift backed rpc agent implementation till then.

Differential Revision: [D16825072](https://our.internmc.facebook.com/intern/diff/D16825072/)

mrshenli reviewed (approved):
LGTM! Thanks!

CI test failures are irrelevant but lint errors are real. Please fix lint before land.

CI:

```
Aug 28 01:59:46 =================================== FAILURES ===================================
Aug 28 01:59:46 _ TestONNXRuntime_opset11.test_elman_tanh_trilayer_forward_with_initial_state_without_sequence_lengths_without_dropout _
Aug 28 01:59:46 
Aug 28 01:59:46 self = <test_pytorch_onnx_onnxruntime.TestONNXRuntime_opset11 testMethod=test_elman_tanh_trilayer_forward_with_initial_state_without_sequence_lengths_without_dropout>
Aug 28 01:59:46 
Aug 28 01:59:46     def wrapper(self):
Aug 28 01:59:46         if self.opset_version < min_opset_version:
Aug 28 01:59:46             raise unittest.SkipTest("Skip verify test for unsupported opset_version")
Aug 28 01:59:46 >       return func(self)
Aug 28 01:59:46 
Aug 28 01:59:46 test/onnx/test_pytorch_common.py:54: 
```

Lint:

```
        return createException(std::move(request), e);
                               ^~~~~~~~~~       ~
```

satgera committed:
Update on "Return a message instead of void from rpc udf"


Return a message instead of void from rpc udf
This is to help thrift style rpc where there is no need for explicit send for a response.
We need to figure out how to solve the non-blocking callback case but don't want to block the thrift backed rpc agent implementation till then.

Differential Revision: [D16825072](https://our.internmc.facebook.com/intern/diff/D16825072/)
